### PR TITLE
Added a test Case for the recursion error in `function_range`

### PR DESCRIPTION
**Pull request by samithkavishke**
#### References to other Issues or PRs

Partial fix for #26076 

#### Brief description of what is fixed or changed
I have added a simple test case to validate this case

#### Other comments
None

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/calculus/tests/test_util.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ def test_function_range():
         ) == Interval(-1, 1)
     assert function_range(cos(x), x, S.EmptySet) == S.EmptySet
     assert function_range(x/sqrt(x**2+1), x, S.Reals) == Interval.open(-1,1)
+    assert function_range(tan(x)**2 + tan(3*x)**2 + 1, x, S.Reals) == Interval(1,oo)
```
Comment (oscarbenjamin) [inline sympy/calculus/tests/test_util.py:60]: This should be put in a separate test function marked `@slow` because it takes over 10 seconds.

Comment (samithkavishke) [reply sympy/calculus/tests/test_util.py:60]: okay I will do that

File: sympy/calculus/tests/test_util.py [hunk 2]
```diff
@@ -66,6 +66,10 @@ def test_function_range():
     raises(NotImplementedError, lambda : function_range(
         sin(x)/2, x, S.Naturals))
 
+@slow
+def test_function_range1():
+    assert function_range(tan(x)**2 + tan(3*x)**2 + 1, x, S.Reals) == Interval(1,oo)
+
 
 def test_continuous_domain():
```
Comment (oscarbenjamin) [inline sympy/calculus/tests/test_util.py:74]: There should be two blank lines between top-level functions.

Comment (samithkavishke) [reply sympy/calculus/tests/test_util.py]: I corrected that

**Pull request merged**